src/LibVLCSharp.Eto/VideoView.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Runtime.InteropServices;
+using Eto.Drawing;
+using Eto.Forms;
+using LibVLCSharp.Shared;
+
+namespace LibVLCSharp.Eto
+{
+    /// <summary>
+    /// Eto.Forms VideoView control with a LibVLCSharp MediaPlayer
+    /// </summary>
+    public class VideoView : Panel, IVideoView, IDisposable
+    {
+        /// <summary>
+        /// The VideoView constructor.
+        /// </summary>
+        public VideoView()
+        {
+            BackgroundColor = Colors.Black;
+        }
+
+        MediaPlayer? _mp;
+
+        /// <summary>
+        /// The MediaPlayer attached to this view (or null)
+        /// </summary>
+        public MediaPlayer? MediaPlayer
+        {
+            get => _mp;
+            set
+            {
+                if (ReferenceEquals(_mp, value))
+                {
+                    return;
+                }
+
+                MpDetach();
+                _mp = value;
+                MpAttach();
+            }
+        }
+
+        void MpDetach()
+        {
+            if (_mp == null || _mp.NativeReference == IntPtr.Zero)
+                return;
+
+            _mp.Hwnd = _mp.NsObject = IntPtr.Zero;
```

Comment:
It may be harmless but we want to avoid all unnecessary native calls to libvlc. On mac, this calls into code like

```
void libvlc_media_player_set_nsobject( libvlc_media_player_t *p_mi,
                                        void * drawable )
{
    assert (p_mi != NULL);
#ifdef __APPLE__
    var_SetString (p_mi, "dec-dev", "any");
    var_SetString (p_mi, "vout", "any");
    var_SetString (p_mi, "window", "any");
    var_SetAddress (p_mi, "drawable-nsobject", drawable);
#else
    (void)drawable;
    libvlc_printerr ("can't set nsobject: APPLE build required");
    assert(false);
    var_SetString (p_mi, "window", "none");
#endif
}
```

`if (RuntimeInformation.IsOSPlatform(OSPlatform.X))` should be used here as well to avoid calling platform-specific function on any platform.